feat: allow disable color vars, radius, global styles

NamesMT opened the pull request:
<!-- DO NOT IGNORE THE TEMPLATE!

Thank you for contributing!

Before submitting the PR, please make sure you do the following:

- Read the [Contributing Guide](https://github.com/antfu/contribute).
- Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- Ideally, include relevant tests that fail without this PR but pass with it.

-->

### Description
This would allow user to config the color variables in css file, as well as finer control over the global styles behavior

### Linked Issues
Related / Resolves #3

### Additional context

<!-- e.g. is there anything you'd like reviewers to focus on? -->

hyoban commented on src/types.ts:
```diff
@@ -18,7 +18,7 @@ export type ThemeOptions = {
    * @default 'zinc'
    */
-  color?: ColorOptions
+  color?: ColorOptions | null
```
Maybe use `false`?

NamesMT replied:
Sure, I think I'll also add false for radius?

hyoban replied:
Yes, It's OK

hyoban commented on src/generate.ts:
```diff
@@ -36,16 +57,16 @@ function getColorTheme(color: ColorOptions) {
   if (typeof color === 'string') {
     name = color
-    ;({ light, dark } = getBuiltInTheme(color))
+    ; ({ light, dark } = getBuiltInTheme(color))
```
Can you help me remove these formatting changes?

NamesMT replied:
Okay, that wasn't expected haha, might be a bit sleepy yesterday

NamesMT commented:
Hi @hyoban, I just force pushed the PR with the following changes:
+ Change `null` to `false`
+ Also allowing to disable `radius` with `false` when in combination with `color`
+ DRYing `colorCSSVarsStyles`'s return
+ Rebased to `main` to resolve conflict

hyoban commented on src/index.ts:
```diff
@@ -21,14 +25,7 @@ export function presetShadcn(options: PresetShadcnOptions = {}): Preset<Theme> {
-            color: hsl(var(--foreground));
-            background: hsl(var(--background));
-          }
+          ${globals ? generateGlobalStyles() : ''}
```
I think maybe we don't need this option. It’s always good to have this.

NamesMT replied:
Not sure about that, it's always better to allow more control, yesterday I was setting up `PrimeVue tailwind preset mode` + `Shadcn` using UnoCSS, and I was wondering what overridden my background color as I did set it in my `main.(s)css` file.
UnoCSS styles (`__uno.css`) is generated after Nuxt's `css` config and could introduce some priority headaches for user control.

Some image for reference:
![image](https://github.com/hyoban/unocss-preset-shadcn/assets/23612546/2eaf06e7-ad03-4c9c-b4eb-9f95c53f6c58)
![image](https://github.com/hyoban/unocss-preset-shadcn/assets/23612546/69a6dd9c-0ec0-4b9d-8475-c886b2412229)

hyoban commented on src/index.ts:
```diff
@@ -1,14 +1,18 @@
+ * @param globals Generates global variables, like *.border-color, body.color, body.background.
+ * @default true
+ */
+export function presetShadcn(options: PresetShadcnOptions = {}, globals?: boolean): Preset<Theme> {
```
Missing default option?

NamesMT commented:
@hyoban I have added the missing default for globals and added tests.
I also moved test snapshots to a sub directory and ignore them in eslint, is that ok?

hyoban commented:
LGTM, Thank you so much for your contribution.

hyoban commented:
Released at 0.3.0, you can give it a try.
